raiden/smart_contracts/TokenNetwork.sol

```diff
@@ -0,0 +1,697 @@
+        );
+
+        // This will reset the transferred amount, invalidating any unlocked locks that were
+        // unlocked but not included in the new locksroot
+        updateParticipantStruct(
+            channel_identifier,
+            closing_participant,
+            nonce,
+            locksroot,
+            transferred_amount
+        );
+
+        TransferUpdated(channel_identifier, closing_participant);
+    }
+
+    /// @notice Registers the lock secret in the SecretRegistry contract. Unlocks a pending transfer and increases the partner's transferred amount
+    /// with the transfer value. A lock can be unlocked only once per participant.
+    // Anyone can call unlock a transfer on behalf of a channel participant.
+    /// @param channel_identifier The channel identifier - mapping key used for `channels`.
+    /// @param partner Address of the participant who owes the locked tokens.
+    /// @param expiration_block Block height at which the lock expires.
+    /// @param locked_amount Amount of tokens that the locked transfer values.
+    /// @param hashlock The hash of the secret
+    /// @param merkle_proof The merkle proof needed to compute the merkle root.
+    /// @param secret A value used as a preimage in a HTL Transfer
+    function registerSecretAndUnlock(
+        uint256 channel_identifier,
+        address partner,
+        uint64 expiration_block,
+        uint256 locked_amount,
+        bytes32 hashlock,
+        bytes merkle_proof,
+        bytes32 secret)
+        stillTimeout(channel_identifier)
+        external
+    {
+        registerSecret(secret);
+        unlock(
+            channel_identifier,
+            partner,
+            expiration_block,
+            locked_amount,
+            hashlock,
+            merkle_proof,
+            secret
+        );
+    }
+
+    /// @notice Registers the lock secret in the SecretRegistry contract.
+    function registerSecret(bytes32 secret) public {
+        require(secret_registry.registerSecret(secret));
+    }
+
+    /// @notice Unlocks a pending transfer and increases the partner's transferred amount
+    /// with the transfer value. A lock can be unlocked only once per participant.
+    // Anyone can call unlock a transfer on behalf of a channel participant.
+    /// @param channel_identifier The channel identifier - mapping key used for `channels`.
+    /// @param partner Address of the participant who owes the locked tokens.
+    /// @param expiration_block Block height at which the lock expires.
+    /// @param locked_amount Amount of tokens that the locked transfer values.
+    /// @param hashlock The hash of the secret
+    /// @param merkle_proof The merkle proof needed to compute the merkle root.
+    /// @param secret A value used as a preimage in a HTL Transfer
+    function unlock(
+        uint256 channel_identifier,
+        address partner,
+        uint64 expiration_block,
+        uint256 locked_amount,
+        bytes32 hashlock,
```

Comment:
the hashlock is not required here, since the secret is provided.

Reply:
Yes, totally right. Will change.